File: librarymanagement/src/main/java/com/example/librarymanagement/service/MemberService.java
package com.example.librarymanagement.service;


import com.example.librarymanagement.dto.MemberDto;
import com.example.librarymanagement.dto.MemberRequest;

import java.util.List;

public interface MemberService {

    List<MemberDto> getAllMembers();

    MemberDto getMemberById(Long id);

    MemberDto createMember(MemberRequest request);

    MemberDto updateMember(Long id, MemberRequest request);

    MemberDto deactivateMember(Long id);
}

File: frontend/src/app/members/services/member.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Member, MemberRequest } from '../models/member';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class MemberService {

  private apiUrl = `${environment.apiUrl}/members`;

  constructor(private http: HttpClient) {}

  getAllMembers(): Observable<Member[]> {
    return this.http.get<Member[]>(this.apiUrl);
  }

  getMemberById(id: number): Observable<Member> {
    return this.http.get<Member>(`${this.apiUrl}/${id}`);
  }

  createMember(member: MemberRequest): Observable<Member> {
    return this.http.post<Member>(this.apiUrl, member);
  }

  updateMember(id: number, member: MemberRequest): Observable<Member> {
    return this.http.put<Member>(`${this.apiUrl}/${id}`, member);
  }

  deactivateMember(id: number): Observable<Member> {
    return this.http.patch<Member>(`${this.apiUrl}/${id}/deactivate`, {});
  }
}